feat: add guard for API Key subscriptions in shared mode

a-cordier opened the pull request:
see https://github.com/gravitee-io/issues/issues/7248
<!-- UI placeholder -->
🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/7248-feat-shared-api-key-subscription-gards/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)
<!-- UI placeholder end -->
<!-- Storybook placeholder -->
---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-awaxryhdro.chromatic.com)
<!-- Storybook placeholder end -->

a-cordier committed:
feat: add guard for API Key subscriptions in shared mode

see https://github.com/gravitee-io/issues/issues/7248

marcambier commented on gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java:
```diff
@@ -250,6 +252,26 @@ public SubscriptionEntity create(NewSubscriptionEntity newSubscriptionEntity, St
             }
 
+            if (applicationEntity.getApiKeyMode() == SHARED && planEntity.getSecurity().equals(API_KEY)) {
+                SubscriptionCriteria criteria = new SubscriptionCriteria.Builder()
```
Subscriptions are already read from repository at line 275.

I wonder if we could use the already exiting `onlyValidSubs` to do this check, for example like it's done for JWT at line 300.
That would avoid an additional repository call.
WDYT ?

a-cordier committed:
fix: console error on readable api subscriptions creation

a-cordier committed:
feat: do not prompt for mode on ambiguous API key subscriptions

We cannot support multiple API key plans for the same API in shared mode,
so there is no need to prompt in that case

marcambier commented on gravitee-apim-console-webui/src/management/application/details/subscribe/application-subscribe.controller.ts:
```diff
@@ -164,19 +164,24 @@ class ApplicationSubscribeController {
   }
+
+  canShareKey(plan: any): boolean {
+    return !this.subscriptions.data.some((subscription) => subscription.api === plan.api && subscription.security === ApiKeyMode.SHARED);
```
```suggestion
    return !this.subscriptions.data.some((subscription) => subscription.api === plan.api && subscription.security ===  PlanSecurityType.API_KEY);
```

marcambier commented on gravitee-apim-console-webui/src/management/application/details/subscribe/application-subscribe.controller.ts:
```diff
@@ -164,19 +164,24 @@ class ApplicationSubscribeController {
     return this.Constants.env?.settings?.plan?.security?.sharedApiKey?.enabled;
   }
+
+  canShareKey(plan: any): boolean {
```
Could we improve readability with some kind of renaming ?
Suggestion : `hasAlreadySubscribedApiKeyPlanOnApi(api)`

a-cordier committed:
fix: application subscription prompt condition
